lib/pyinotify.py

```diff
@@ -0,0 +1,2346 @@
+
+
+class Watch:
+    """
+    Represent a watch, i.e. a file or directory being watched.
+
+    """
+    __slots__ = ('wd', 'path', 'mask', 'proc_fun', 'auto_add',
+                 'exclude_filter', 'dir')
+
+    def __init__(self, wd, path, mask, proc_fun, auto_add, exclude_filter):
+        """
+        Initializations.
+
+        @param wd: Watch descriptor.
+        @type wd: int
+        @param path: Path of the file or directory being watched.
+        @type path: str
+        @param mask: Mask.
+        @type mask: int
+        @param proc_fun: Processing callable object.
+        @type proc_fun:
+        @param auto_add: Automatically add watches on new directories.
+        @type auto_add: bool
+        @param exclude_filter: Boolean function, used to exclude new
+                               directories from being automatically watched.
+                               See WatchManager.__init__
+        @type exclude_filter: callable object
+        """
+        self.wd = wd
+        self.path = path
+        self.mask = mask
+        self.proc_fun = proc_fun
+        self.auto_add = auto_add
+        self.exclude_filter = exclude_filter
+        self.dir = os.path.isdir(self.path)
+
+    def __repr__(self):
+        """
+        @return: String representation.
+        @rtype: str
+        """
+        s = ' '.join(['%s%s%s' % (output_format.field_name(attr),
+                                  output_format.punctuation('='),
+                                  output_format.field_value(getattr(self,
+                                                                    attr))) \
+                      for attr in self.__slots__ if not attr.startswith('_')])
+
+        s = '%s%s %s %s' % (output_format.punctuation('<'),
+                            output_format.class_name(self.__class__.__name__),
+                            s,
+                            output_format.punctuation('>'))
+        return s
+
+
+class ExcludeFilter:
+    """
+    ExcludeFilter is an exclusion filter.
+    """
+    def __init__(self, arg_lst):
+        """
+        Examples:
+          ef1 = ExcludeFilter(["/etc/rc.*", "/etc/hostname"])
+          ef2 = ExcludeFilter("/my/path/exclude.lst")
+          Where exclude.lst contains:
+          /etc/rc.*
+          /etc/hostname
+
+        Note: it is not possible to exclude a file if its encapsulating
+              directory is itself watched. See this issue for more details
+              https://github.com/seb-m/pyinotify/issues/31
+
+        @param arg_lst: is either a list of patterns or a filename from which
+                        patterns will be loaded.
+        @type arg_lst: list of str or str
+        """
+        if isinstance(arg_lst, str):
+            lst = self._load_patterns_from_file(arg_lst)
+        elif isinstance(arg_lst, list):
+            lst = arg_lst
+        else:
+            raise TypeError
+
+        self._lregex = []
+        for regex in lst:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)